src/custom/GoogleDLPRedactor.ts

```diff
@@ -107,6 +110,66 @@ const likelihoodPriority: { [likelyHoodName: string]: number } = {
 
 const includeQuote = true;
 
+interface Finding {
+  likelihood: string;
+  quote: string;
+  infoType: {
+    name: string;
+  };
+  location: {
+    byteRange: {
+      start: string;
+      end: string;
+    };
+  };
+}
+
+/**
+ * Remove overlapping findings which can cause messed up tokens.
+ *
+ * For example "My name is John D." will cause 3 findings:
+ *  - PERSON_NAME for text "John S." at range 11-17
+ *  - FIRST_NAME for text "John" at range 11-15
+ *  - LAST_NAME for text "S." at range 15-17
+ *
+ * The FIRST_NAME and LAST_NAME findings overlap the first finding so there is no need to search for them
+ */
+function removeOverlappingFindings(findings: Finding[]): Finding[] {
+  // early return if only have 0 or 1 findings
+  if (findings.length <= 1) {
+    return findings;
+  }
+
+  // finding location.byteRange.start and end are strings for some reason, so must convert to numbers
+  const getStart = (finding: Finding) => Number(get(finding, 'location.byteRange.start', 0));
+  const getEnd = (finding: Finding) => Number(get(finding, 'location.byteRange.end', 0));
+
+  // sort findings by ascending start and store sorted indices in new array
+  const findingIndicesOrderedByStart = findings.map((finding, index) => index);
```

Comment:
why do you create this second array of indices? You can simply sort the existing findings array instead of an array of it's indices

Reply:
I was trying to leave the original array untouched, but I realize now that was only needed in an earlier version of the implementation.  Since the current version doesn't require any pre-existing sort, I've moved the likelihood sorting to after the overlaps are removed. Then I've simplified the removeOverlappingFindings to do the sorting in place.